2025-02-24 13:32:27,532 - app.utils.opcua_client - INFO - Connecting to OPCUA server at opc.tcp://localhost:4841
2025-02-24 13:32:29,592 - app.utils.opcua_client - INFO - Connected to OPCUA server at opc.tcp://localhost:4841
2025-02-24 13:32:29,593 - app.main - INFO - Successfully connected to OPCUA server
2025-02-24 13:32:30,101 - app.routes.websocket - INFO - WebSocket connection established
2025-02-24 13:32:30,101 - app.utils.opcua_client - INFO - Connecting to OPCUA server at opc.tcp://localhost:4841
2025-02-24 13:32:32,164 - app.utils.opcua_client - INFO - Connected to OPCUA server at opc.tcp://localhost:4841
2025-02-24 13:32:32,170 - app.utils.opcua_client - INFO - Created OPCUA subscription
2025-02-24 13:32:32,687 - app.routes.websocket - INFO - Sent WebSocket update: {'variable1': 0}
2025-02-24 13:32:32,688 - app.routes.websocket - INFO - Sent WebSocket update: {'variable2': 'initial_state'}
2025-02-24 13:32:32,690 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 20.449945537681664}
2025-02-24 13:32:33,187 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 17.34925046611469}
2025-02-24 13:32:34,187 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 21.282399216802105}
2025-02-24 13:32:35,174 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 22.218506501360736}
2025-02-24 13:32:36,186 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 19.98240621548824}
2025-02-24 13:32:37,156 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 19.482156358227197}
2025-02-24 13:32:38,182 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 24.57120554690018}
2025-02-24 13:32:39,667 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 22.328745125777196}
2025-02-24 13:32:40,661 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 16.57283771363814}
2025-02-24 13:32:41,677 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 20.29241458576162}
2025-02-24 13:32:42,678 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 24.585213375840894}
2025-02-24 13:32:43,676 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 18.64547746716116}
2025-02-24 13:32:44,679 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 21.467174406692475}
2025-02-24 13:32:45,666 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 16.140111931371674}
2025-02-24 13:32:46,184 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 20.550318802221057}
2025-02-24 13:32:47,625 - app.routes.websocket - INFO - WebSocket connection established
2025-02-24 13:32:47,632 - app.utils.opcua_client - INFO - Connecting to OPCUA server at opc.tcp://localhost:4841
2025-02-24 13:32:47,634 - app.utils.opcua_client - INFO - Created OPCUA subscription
2025-02-24 13:32:47,671 - app.routes.websocket - ERROR - Error in datachange_notification: 
2025-02-24 13:32:48,128 - app.routes.websocket - INFO - Sent WebSocket update: {'variable1': 0}
2025-02-24 13:32:48,129 - app.routes.websocket - INFO - Sent WebSocket update: {'variable2': 'initial_state'}
2025-02-24 13:32:48,130 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 20.121166466520144}
2025-02-24 13:32:48,646 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 19.25677389177074}
2025-02-24 13:32:48,675 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:32:49,647 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 15.434998287330336}
2025-02-24 13:32:49,651 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:32:49,665 - app.utils.opcua_client - INFO - Connected to OPCUA server at opc.tcp://localhost:4841
2025-02-24 13:32:49,672 - app.routes.data - INFO - Retrieved data: {'variable1': 0, 'variable2': 'initial_state', 'Temperature': 15.434998287330336}
2025-02-24 13:32:49,681 - app.routes.data - INFO - Retrieved data: {'variable1': 0, 'variable2': 'initial_state', 'Temperature': 15.434998287330336}
2025-02-24 13:32:50,195 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:32:50,649 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 21.136028720368163}
2025-02-24 13:32:51,651 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 22.627535776484358}
2025-02-24 13:32:51,655 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:32:52,636 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 18.97931133541581}
2025-02-24 13:32:52,668 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:32:53,618 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 22.15518128922971}
2025-02-24 13:32:53,686 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:32:54,645 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 16.878396207012674}
2025-02-24 13:32:54,676 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:32:55,645 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 21.335099903938314}
2025-02-24 13:32:55,676 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:32:56,644 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 19.236419067408534}
2025-02-24 13:32:56,675 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:32:57,645 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 18.805121458782196}
2025-02-24 13:32:57,693 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:32:58,641 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 24.83942353297789}
2025-02-24 13:32:58,672 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:32:59,642 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 15.688374242613088}
2025-02-24 13:32:59,688 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:00,638 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 16.435465056364226}
2025-02-24 13:33:00,684 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:01,636 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 20.954117905287795}
2025-02-24 13:33:01,685 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:02,648 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 17.23273849577897}
2025-02-24 13:33:02,681 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:03,646 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 20.44415997958847}
2025-02-24 13:33:03,679 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:04,646 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 24.611875766156697}
2025-02-24 13:33:04,661 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:05,642 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 23.49198022721793}
2025-02-24 13:33:05,673 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:06,641 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 18.357810566987602}
2025-02-24 13:33:06,686 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:07,639 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 17.173006575000226}
2025-02-24 13:33:07,686 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:08,646 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 15.588631460037758}
2025-02-24 13:33:08,691 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:09,642 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 15.36341849396482}
2025-02-24 13:33:09,674 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:10,642 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 24.69898381840933}
2025-02-24 13:33:10,687 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:11,651 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 20.749698594914236}
2025-02-24 13:33:11,680 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:12,634 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 21.898055342938612}
2025-02-24 13:33:12,666 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:13,646 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 20.564443301559827}
2025-02-24 13:33:13,677 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:14,644 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 16.367326467104153}
2025-02-24 13:33:14,674 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:15,640 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 22.005768510192958}
2025-02-24 13:33:15,684 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:16,652 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 23.613529708276715}
2025-02-24 13:33:16,685 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:17,650 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 19.095382669274276}
2025-02-24 13:33:17,681 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:18,650 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 16.093098776349926}
2025-02-24 13:33:18,654 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:19,645 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 21.240739767080846}
2025-02-24 13:33:19,675 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:20,645 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 16.87638284421576}
2025-02-24 13:33:20,689 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:21,637 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 19.033891662346477}
2025-02-24 13:33:21,684 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:22,634 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 21.010760748479882}
2025-02-24 13:33:22,681 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:23,621 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 19.49001354139303}
2025-02-24 13:33:23,683 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:24,649 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 16.291703332693633}
2025-02-24 13:33:24,680 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:25,646 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 19.80259780211148}
2025-02-24 13:33:25,689 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:26,641 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 22.556173378311826}
2025-02-24 13:33:26,673 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:27,636 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 18.60523640821083}
2025-02-24 13:33:27,684 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:28,638 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 15.52439793309541}
2025-02-24 13:33:28,668 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:29,636 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 17.434300419953924}
2025-02-24 13:33:29,682 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:30,649 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 18.21132205025966}
2025-02-24 13:33:30,681 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:31,636 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 16.954569946444547}
2025-02-24 13:33:31,683 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:32,637 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 24.545722761765973}
2025-02-24 13:33:32,683 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:33,633 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 15.936584553182723}
2025-02-24 13:33:33,681 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:34,634 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 21.664051175611057}
2025-02-24 13:33:34,682 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:35,645 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 19.716076473782877}
2025-02-24 13:33:35,677 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:36,640 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 20.60209291571728}
2025-02-24 13:33:36,672 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:37,646 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 21.161858508192925}
2025-02-24 13:33:37,675 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:38,639 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 17.13831004115381}
2025-02-24 13:33:38,683 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:39,664 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 16.718772387361266}
2025-02-24 13:33:39,668 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:40,642 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 18.165067509761894}
2025-02-24 13:33:40,672 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:41,646 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 18.78063995595845}
2025-02-24 13:33:41,692 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:42,632 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 18.03955493658275}
2025-02-24 13:33:42,662 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:43,645 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 15.1118035381576}
2025-02-24 13:33:43,691 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:44,626 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 18.76253611923858}
2025-02-24 13:33:44,686 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:45,636 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 16.16051317657268}
2025-02-24 13:33:45,683 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:46,635 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 23.076909916086002}
2025-02-24 13:33:46,681 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:47,632 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 16.198216571883258}
2025-02-24 13:33:47,661 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:48,644 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 21.21233416692248}
2025-02-24 13:33:48,686 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:49,684 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:50,139 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 16.79796103940162}
2025-02-24 13:33:50,684 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:51,142 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 19.48436167197282}
2025-02-24 13:33:51,654 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:52,149 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 15.970544893615763}
2025-02-24 13:33:52,668 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:53,152 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 16.47194930147699}
2025-02-24 13:33:53,682 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:54,150 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 15.667049254048958}
2025-02-24 13:33:54,679 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:55,147 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 20.654093686378765}
2025-02-24 13:33:55,677 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:56,143 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 21.574643422298404}
2025-02-24 13:33:56,677 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:57,146 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 19.785889073155705}
2025-02-24 13:33:57,680 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:58,146 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 23.956443283891232}
2025-02-24 13:33:58,678 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:33:59,144 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 20.132439827814125}
2025-02-24 13:33:59,676 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:34:00,149 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 20.51500800141199}
2025-02-24 13:34:00,692 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:34:01,144 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 16.57967743939082}
2025-02-24 13:34:01,635 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 17.9999583110927}
2025-02-24 13:34:01,678 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:34:02,632 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 20.965298296190067}
2025-02-24 13:34:02,661 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:34:03,640 - app.routes.websocket - INFO - Sent WebSocket update: {'Temperature': 16.573379108493622}
2025-02-24 13:34:03,686 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:34:04,683 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:34:05,151 - app.routes.websocket - ERROR - Error in datachange_notification: 
2025-02-24 13:34:05,667 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:34:06,150 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:34:06,668 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:34:07,149 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:34:07,677 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:34:08,143 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:34:08,677 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:34:09,143 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:34:09,669 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:34:10,137 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:34:10,672 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:34:11,141 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:34:11,685 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:34:12,138 - app.routes.websocket - ERROR - Error in datachange_notification: Cannot call "send" once a close message has been sent.
2025-02-24 13:34:12,600 - app.utils.opcua_client - INFO - Disconnected from OPCUA server
2025-02-24 13:34:12,601 - app.main - INFO - Successfully disconnected from OPCUA server
